modules/solid_mechanics/src/materials/ADComputeIncrementalShellStrain.C

```diff
@@ -388,12 +406,26 @@ ADComputeIncrementalShellStrain::computeGMatrix()
       // _transformation_matrix is an AD property, but we're not setting the derivatives!
       (*_transformation_matrix)[i] = J;
 
-      // calculate ge
-      ADRealVectorValue e3 = (*_dxyz_dzeta[j])[i] / (*_dxyz_dzeta[j])[i].norm();
-      ADRealVectorValue e1 = (*_dxyz_deta[j])[i].cross(e3);
-      e1 /= e1.norm();
-      ADRealVectorValue e2 = e3.cross(e1);
-      e2 /= e2.norm();
+      // compute element's local coordinate
+      computeLocalCoordinates();
+
+      // calculate the local transformation matrix to be used to map the global stresses to the
+      // local element coordinate
+      ADRankTwoTensor local_rotation_mat;
+      local_rotation_mat(0, 0) = _e1(0);
+      local_rotation_mat(0, 1) = _e1(1);
+      local_rotation_mat(0, 2) = _e1(2);
+      local_rotation_mat(1, 0) = _e2(0);
+      local_rotation_mat(1, 1) = _e2(1);
+      local_rotation_mat(1, 2) = _e2(2);
+      local_rotation_mat(2, 0) = _e3(0);
+      local_rotation_mat(2, 1) = _e3(1);
+      local_rotation_mat(2, 2) = _e3(2);
```

Comment:
here we have a "from rows" named constructor to make this a one liner

Reply:
done;)